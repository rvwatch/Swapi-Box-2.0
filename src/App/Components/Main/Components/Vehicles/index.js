import React, { Component } from 'react';
import { array, func } from 'prop-types';
import Card from '../Card';

class Vehicles extends Component {
  render() {
    const createCards = this.props.vehicles.map((vehicle, index) => {
      const favIds = this.props.favorites.map(fav => fav.id);
      const favStatus = favIds.includes(vehicle.id);
      return (
        <Card
          toggleFavorites={this.props.toggleFavorites}
          cardData={vehicle}
          isFav={favStatus}
          {...vehicle}
          key={vehicle.name + index}
        />
      );
    });

    return <section>{createCards}</section>;
  }
}

Vehicles.propTypes = {
  vehicles: array,
  planets: array,
  starships: array,
  favorites: array,
  toggleFavorites: func
};

export default Vehicles;
