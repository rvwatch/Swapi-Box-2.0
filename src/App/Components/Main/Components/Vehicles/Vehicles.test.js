import React from 'react';
import { shallow } from 'enzyme';
import Vehicles from './index';
import {
  mockCleanVehicle,
  mockFavorites
} from '../../../../../apiCalls/MockedData';

describe('Vehicles ', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Vehicles favorites={mockFavorites} vehicles={mockCleanVehicle} />
    );
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
