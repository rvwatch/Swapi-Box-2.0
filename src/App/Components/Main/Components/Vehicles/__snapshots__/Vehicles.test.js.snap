// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vehicles  renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Vehicles
    favorites={
        Array [
            Object {
              "homeworld": "Tatooine",
              "id": 1,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 2,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "id": 3,
              "name": "Luke Skywalker",
              "population": 200000,
              "species": "Human",
            },
          ]
    }
    vehicles={
        Array [
            Object {
              "id": "Sand Crawler0",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicle_class": "wheeled",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          vehicles cards
</h1>,
        Array [
          <Card
            cardData={
                        Object {
                                    "id": "Sand Crawler0",
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                    "passengers": "30",
                                    "vehicle_class": "wheeled",
                                  }
            }
            id="Sand Crawler0"
            isFav={false}
            model="Digger Crawler"
            name="Sand Crawler"
            passengers="30"
            toggleFavorites={undefined}
            vehicle_class="wheeled"
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "vehicles cards",
        },
        "ref": null,
        "rendered": "vehicles cards",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "Sand Crawler0",
        "nodeType": "function",
        "props": Object {
          "cardData": Object {
            "id": "Sand Crawler0",
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
            "vehicle_class": "wheeled",
          },
          "id": "Sand Crawler0",
          "isFav": false,
          "model": "Digger Crawler",
          "name": "Sand Crawler",
          "passengers": "30",
          "toggleFavorites": undefined,
          "vehicle_class": "wheeled",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            vehicles cards
</h1>,
          Array [
            <Card
              cardData={
                            Object {
                                          "id": "Sand Crawler0",
                                          "model": "Digger Crawler",
                                          "name": "Sand Crawler",
                                          "passengers": "30",
                                          "vehicle_class": "wheeled",
                                        }
              }
              id="Sand Crawler0"
              isFav={false}
              model="Digger Crawler"
              name="Sand Crawler"
              passengers="30"
              toggleFavorites={undefined}
              vehicle_class="wheeled"
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "vehicles cards",
          },
          "ref": null,
          "rendered": "vehicles cards",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "Sand Crawler0",
          "nodeType": "function",
          "props": Object {
            "cardData": Object {
              "id": "Sand Crawler0",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
              "vehicle_class": "wheeled",
            },
            "id": "Sand Crawler0",
            "isFav": false,
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
            "toggleFavorites": undefined,
            "vehicle_class": "wheeled",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
